packages/connect-node/src/grpc-http-transport.ts

```diff
@@ -148,16 +171,34 @@ export function createGrpcHttpTransport(
             method,
             url: createMethodUrl(options.baseUrl, service, method),
             init: {},
-            header: grpcCreateRequestHeader(useBinaryFormat, timeoutMs, header),
+            header: grpcCreateRequestHeaderWithCompression(
+              method.kind,
+              useBinaryFormat,
+              timeoutMs,
+              header,
+              acceptCompression.map((c) => c.name),
+              options.sendCompression?.name
+            ),
             message: normalize(message),
             signal: signal ?? new AbortController().signal,
           },
           async (req: UnaryRequest<I>): Promise<UnaryResponse<O>> => {
-            const envelope = encodeEnvelope(
-              messageFlag,
-              serialize(req.message)
-            );
             const encoding = useBinaryFormat ? "binary" : "utf8";
+            let flag = messageFlag;
```

Comment:
The line above is:

```ts
const encoding = useBinaryFormat ? "binary" : "utf8";
```

We pass the encoding to `makeNodeRequest()` and method `write()` of `http.ClientRequest`.

But with compression, we only ever write Uint8Array. Please remove the variable and code in `makeNodeRequest` completely from all transports.